content/docs/context.md

```diff
@@ -4,220 +4,92 @@ title: Context
-
-- [`constructor(props, context)`](/docs/react-component.html#constructor)
-- [`componentWillReceiveProps(nextProps, nextContext)`](/docs/react-component.html#componentwillreceiveprops)
-- [`shouldComponentUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#shouldcomponentupdate)
-- [`componentWillUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#componentwillupdate)
+Requires a [function as a child](/docs/render-props.html#using-props-other-than-render). This function receives the current context value and returns a React node. It will be called whenever the Provider's value is updated.
 
 > Note:
->
-> As of React 16, `componentDidUpdate` no longer receives `prevContext`.
-
-## Referencing Context in Stateless Functional Components
-
-Stateless functional components are also able to reference `context` if `contextTypes` is defined as a property of the function. The following code shows a `Button` component written as a stateless functional component.
-
-```javascript
-import PropTypes from 'prop-types';
+> 
+> For more information about this pattern, see [render props](/docs/render-props.html).
 
-const Button = ({children}, context) =>
-  <button style={{background: context.color}}>
-    {children}
-  </button>;
+## Examples
 
-Button.contextTypes = {color: PropTypes.string};
-```
-
-## Updating Context
-
-Don't do it.
+### Static Context
 
-React has an API to update context, but it is fundamentally broken and you should not use it.
+Here is an example illustrating how you might inject a "theme" using context:
 
-The `getChildContext` function will be called when the state or props changes. In order to update data in the context, trigger a local state update with `this.setState`. This will trigger a new context and changes will be received by the children.
+`embed:context/theme-example.js`
 
-```javascript
-import PropTypes from 'prop-types';
+### Dynamic Context
 
-class MediaQuery extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {type:'desktop'};
-  }
+A more complex example with dynamic values for the theme:
 
-  getChildContext() {
-    return {type: this.state.type};
-  }
+**theme-context.js**
+`embed:context/theme-detailed-theme-context.js`
 
-  componentDidMount() {
-    const checkMediaQuery = () => {
-      const type = window.matchMedia("(min-width: 1025px)").matches ? 'desktop' : 'mobile';
-      if (type !== this.state.type) {
-        this.setState({type});
-      }
-    };
+**themed-button.js**
+`embed:context/theme-detailed-themed-button.js`
 
-    window.addEventListener('resize', checkMediaQuery);
-    checkMediaQuery();
-  }
+**app.js**
+`embed:context/theme-detailed-app.js`
 
-  render() {
-    return this.props.children;
-  }
-}
-
-MediaQuery.childContextTypes = {
-  type: PropTypes.string
-};
-```
```

Comment:
There needs to be a section on [how to reference context in lifecycle methods](https://github.com/facebook/react/issues/12397#issuecomment-375501574). It can be deduced from this article but people will initially feel it's gross so we need to be explicit that it's a recommended pattern. Additionally it's important to show how it works with ref forwarding.

Reply:
There should also be an explanation on how to consume more than one context. Even if it links to a gist.